etc/afpd/status.c

```diff
@@ -548,16 +550,82 @@ void status_init(AFPObj *dsi_obj, AFPObj* asp_obj, DSI *dsi)
     if ( statuslen < maxstatuslen)
         statuslen = status_utf8servername(status, &c, dsi, options);
 
+    dsi->signature = status + sigoff;
+    dsi->statuslen = statuslen;
+
+    /* end dsi status response */
+
 #ifndef NO_DDP
+    /* begin asp status response */
+    char *aspstatus = asp_obj->aspstatus;
+    c = 0; statuslen = 0;
+    status_flags(aspstatus,
+        options->flags& OPTION_SERVERNOTIF,
+        (options->fqdn || ipok),
+        options->passwdbits,
+        0,
+        options->flags);
+    /* returns offset to signature offset */
+    c = status_server(aspstatus, options->hostname, options);
+    status_machine(aspstatus);
+    status_versions(aspstatus,
+        asp,
+        NULL);
+    status_uams(aspstatus, options->uamlist);
+
+    if (strcmp(options->legacyicon, "daemon") == 0) {
+        icon = daemon_icon;
+        iconlen = sizeof(daemon_icon);
+    }
```

Comment:
We get duplication of the icon lookup code here. Better would be to have a shared utility function.